pkg/event/controller.go

```diff
@@ -95,9 +113,30 @@ func (gen *generator) Run(ctx context.Context, workers int, waitGroup *sync.Wait
 }
 
 func (gen *generator) startRecorders(ctx context.Context) error {
-	if err := gen.broadcaster.StartRecordingToSinkWithContext(ctx); err != nil {
+	eventHandler := func(obj runtime.Object) {
+		event, ok := obj.(*apieventsv1.Event)
+		if !ok {
+			gen.logger.Error(nil, "unexpected type, expected eventsv1.Event")
+			return
+		}
+
+		eventCopy := event.DeepCopy()
```

Comment:
why do we need to deep copy the event? is it mutable?

Reply:
Other watchers have access to it too. It's mutable I believe.